### PR TITLE
ERC20 Rewards Wrapper

### PR DESCRIPTION
**Pull request by alcueca**
ERC20 Wrapper for tokens used as a reward on ERC20Rewards, sidestepping the limitation of not being able to swap the rewards token once set.

### CODE REVIEW

File: src/token/ERC20RewardsWrapper.sol [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        uint _totalAssets = asset.balanceOf(address(this));
+        return _totalAssets > 0 ? assets * _totalSupply / _totalAssets : 0;
+    }
+
+    /// @dev A permissioned function that allows to set or change the asset.
+    /// @notice If changing the asset, the caller may consider keeping the price per share constant in USD or ETH terms.
+    /// After changing the asset, use `skim` to remove any old funds.
+    /// Be careful what address you pass on as an ERC20, it is not validated in any way.
+    /// Funds should be transferred before this call. Consider rounding.
+    function set(ERC20 asset_) external virtual auth {
```
Comment (iamsahu) [inline src/token/ERC20RewardsWrapper.sol:50]: I am just a little bit worried about potential rug pulling wrt type of reward our user will be exposed too and if they would look at this favourable.

**Review by iamsahu — COMMENTED** [review]
Apart from a small concern the rest looks good.

File: src/token/ERC20RewardsWrapper.sol (hunk 1 above)
Comment (alcueca) [reply src/token/ERC20RewardsWrapper.sol:50]: Yes, that is true. That should be disclosed to the user as [we disclose all the other ways](https://docs.yieldprotocol.com/#/operations/how_to_hack) in which we can use governance to rug pull.

In mitigation, such a governance action would be subject to a timelock delay, and users would have the opportunity to claim their rewards before the change.

**Review by iamsahu — APPROVED** [review]
LGTM

**Pull request merged**